### PR TITLE
Device integration

### PR DESCRIPTION
**Pull request by bmleight**
Enhanced the library and example app to:

* discover VP3300 devices over bluetooth
* select and connect to a device
* activate the card reader
* read a credit card from the chip reader or swiper

Heavily inspired from the example android app that came with their SDK

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "dependencies": {
     "react": "16.9.0",
     "react-native": "0.61.5",
-    "react-native-card-connect": "git+https://github.com/BigRoomStudios/react-native-card-connect.git#229fd6bce41ba3cbb86ef0197a799eb597e6b5ac"
+    "react-native-card-connect": "git+https://github.com/BigRoomStudios/react-native-card-connect.git#ff462b6e2fa79f47f0551b25a1d05cabed19b037"
```
Comment (bmleight) [inline example/package.json]:
```suggestion
    "react-native-card-connect": "git+https://github.com/BigRoomStudios/react-native-card-connect.git"
```
Commit before merging

File: android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java [hunk 2]
```diff
@@ -1,27 +1,263 @@
...
+
+    @ReactMethod
+    public void addListener(String eventName) {
+        // Set up any upstream listeners or background tasks as necessary
+    }
+
+    @ReactMethod
+    public void removeListeners(Integer count) {
+        // Remove upstream listeners, stop unnecessary background tasks
+    }
```
Comment (devinivy) [inline android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java]: Would be a little interested to see a comment here on why these empty methods are needed.  I presume some other library is calling them— if that's the case and you already know which library it is, maybe just a quick note here about it?

File: android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java [hunk 3]
```diff
@@ -54,8 +291,8 @@ public void onCCConsumerTokenResponse(CCConsumerAccount ccConsumerAccount) {
             });
         } catch (ValidateException e) {
             promise.reject(e);
-        }catch (Exception e){
-            promise.resolve(e);
+        } catch (Exception e) {
+            promise.reject(e);
```
Comment (devinivy) [inline android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java:284]: 😳 good find!

File: package-lock.json [hunk 4]
```diff
@@ -1,8 +1,6857 @@
 {
```
Comment (devinivy) [inline package-lock.json]: While these are useful in applications, I generally just omit package-locks entirely from libraries.  I find it's generally more useful to use the latest deps, as a consuming application might validly be doing.

**Review by devinivy — APPROVED** [review]
Yup!!

File: android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java (hunk 2 above)
Comment (bmleight) [reply android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java]: Good catch!  I plopped these in from somewhere, and then ended up going in a different direction.  Giving them the 🪓

File: android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java (hunk 3 above)
Comment (bmleight) [reply android/src/main/java/com/reactbolt/sdk/RNBoltReactLibraryModule.java:284]: 👃

File: package-lock.json (hunk 4 above)
Comment (bmleight) [reply package-lock.json]: 👍 I modified this accidentally and then forgot to clean up after myself.   I agree, will give it the 🥾

**Pull request merged**